custom_components/imou_life/__init__.py

```
@@ -198,6 +201,31 @@ def _configure_device_options(device: ImouDevice, entry: ConfigEntry):
+                eid for domain, eid in device_entry.identifiers if domain == DOMAIN
+            ]
+            if not imou_ids:
+                continue
+
+            if not any(eid in active_entry_ids for eid in imou_ids):
+                _LOGGER.info(
+                    "Removing orphan device '%s' (no matching config entry)",
+                    device_entry.name,
+                )
+                device_registry.async_remove_device(device_entry.id)
+    except Exception:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)